src/ServiceInsight/MessageHeaders/MessageHeadersView.xaml

```diff
@@ -94,22 +94,22 @@
                 </Trigger>
             </Style.Triggers>
         </Style>
+        <DataTemplate x:Key="TextTemplate">
+            <TextBlock Text="{Binding Value}" d:DataContext="{d:DesignInstance messageHeaders:MessageHeadersKeyValueList}" />
+        </DataTemplate>
+        
+        <DataTemplate x:Key="ExceptionTemplate">
+            <controls:MvvmTextEditor d:DataContext="{d:DesignInstance messageHeaders:MessageHeadersKeyValueList}"
+                                     FontFamily="Consolas"
+                                     FontSize="10pt"
+                                     IsReadOnly="true"
+                                     ScrollViewer.CanContentScroll="False"
+                                     Style="{StaticResource MvvmTextEditorScrollFixStyle}"
+                                     SyntaxHighlighting="StackTrace"
+                                     Text="{Binding Value}" />
+        </DataTemplate>
     </UserControl.Resources>
     <Grid>
-        <Grid.Resources>
-            <DataTemplate x:Key="TextTemplate">
-                <TextBlock Text="{Binding Value}" />
-            </DataTemplate>
-            <DataTemplate x:Key="ExceptionTemplate">
-                <controls:MvvmTextEditor FontFamily="Consolas"
-                                         FontSize="10pt"
-                                         IsReadOnly="true"
-                                         ScrollViewer.CanContentScroll="False"
-                                         Style="{StaticResource MvvmTextEditorScrollFixStyle}"
-                                         SyntaxHighlighting="StackTrace"
-                                         Text="{Binding Value}" />
-            </DataTemplate>
-        </Grid.Resources>
```

Comment:
This is moving the data templates to the upper scope (UserControl). The other issue is that the DesignTime data context doesn't support generics so the ReactiveList<T> is extracted into its own class and used. 

This allows the static analysis tools to find which class/property we have bound to.